### ISSUE TITLE
ShellDetectionFailure for zsh

### ISSUE THREAD

**Issue by pmeier** [sarugaku/shellingham#77]
Not sure what is going on here:

```
❯ echo $SHELL
/usr/bin/zsh
❯ /usr/bin/zsh --version
zsh 5.9 (x86_64-pc-linux-gnu)
❯ python -c 'import shellingham; shellingham.detect_shell()'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/philip/.conda/envs/ora-dev/lib/python3.11/site-packages/shellingham/__init__.py", line 24, in detect_shell
    raise ShellDetectionFailure()
shellingham._core.ShellDetectionFailure
❯ python -c 'import shellingham; print(shellingham.__version__)'
1.5.1
```

**Comment by uranusjr** [comment]
It would be best if you could do some digging since it’s close to impossible to replicate your exact environment. There is no way I can help with just this information.

**Comment by pmeier** [comment]
> It would be best if you could do some digging since it’s close to impossible to replicate your exact environment.

Will do and get back to you.

> There is no way I can help with just this information.

I couldn't find any information about what I should provide. Happy to oblige.

**Comment by pmeier** [comment]
It seems the detection fails, because it doesn't find any processes other than the current one:

```
❯ cat -p git/ora/foo.py
from shellingham import detect_shell

detect_shell()
❯ python git/ora/foo.py
detect_shell(): impl=<module 'shellingham.posix' from '/home/philip/.conda/envs/ora-dev/lib/python3.11/site-packages/shellingham/posix/__init__.py'>
get_shell(): pid='7531', mapping={'7531': Process(args=('python', 'git/ora/foo.py'), pid='7531', ppid='34819')}
get_shell(): proc_args=('python', 'git/ora/foo.py')
Traceback (most recent call last):
  File "/home/philip/git/ora/foo.py", line 3, in <module>
    detect_shell()
  File "/home/philip/.conda/envs/ora-dev/lib/python3.11/site-packages/shellingham/__init__.py", line 25, in detect_shell
    raise ShellDetectionFailure()
shellingham._core.ShellDetectionFailure
```

**Comment by uranusjr** [comment]
Interesting. Can you describe your environment a bit? What operating system are you on, is there emulation involved (32-64 bit binaries, for example), a user that may have permission restrictions, or similar things? Try to dig into doing the process detection part manually as well, does the detection method (proc os ps?) really return only the current process, or is there a problem with parsing?

**Comment by pmeier** [comment]
Looking at the [table](https://docs.kernel.org/filesystems/proc.html#id8) linked in

https://github.com/sarugaku/shellingham/blob/0566a5f5ee35f274a5894285107ee053abf5a99f/src/shellingham/posix/proc.py#L14-L16

I think the numbers are switched? At least the `PPid` is in the seventh row.

**Comment by pmeier** [comment]
Yep, that seems to be it. Will send a PR.

**Comment by uranusjr** [comment]
The version was yanked due to an incompatibility. I attempted to address it and released 1.5.2.

**Comment by pmeier** [comment]
Can confirm that this particular issue is fixed in `1.5.2`. Thanks!
